### PR TITLE
Search package: move search dashboard to package ( the base ) (1/2)

### PR DESCRIPTION
**Pull request by kangzj**
#### Changes proposed in this Pull Request:
The PR is a combination of #21840, #21841 and #21842, which includes the dependencies duplicated from Jetpack and a new state store. The PR doesn't touch the Jetpack the plugin (except for a dependency update) and will serve as a base to migrate the search dashboard on.

As the individual PRs were all reviewed by the search team already ( and most the team is on AFK now ), so I'm sending this for crew review straightly.

#21840 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/a9f40e8d2a448a280005ae2bfce14389b55d7157))
- Copied dependencies from Jetpack the plugin with zero change as a base. tests and state are skipped as we'll add new.

#21841 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/a773e56ba7617ca97ef64b52c7acbfd0e81a36c3))
- Added new state store based on @wordpress/data to replace the state store in Jetpack client. 

#21842 ([commit](https://github.com/Automattic/jetpack/pull/21865/commits/c255b3e24c271c8dfd5c232c0f807e2f1ad040d1))
- Removed unnecessary dependencies from copied components.
- Added simple tests for components
- Make form toggle component grid layout compatible

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Run `(cd projects/packages/search && pnpm test)`
- Run `(cd projects/packages/search && composer phpunit)`
- Ensure all tests pass
- Ensure no changes to Jetpack the plugin except for a dependency version bump

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/button/style.scss [hunk 1]
```diff
@@ -0,0 +1,206 @@
+@import "../../scss/calypso-colors";
+@import "../../scss/color-functions";
+@import "../../scss/functions/rem";
+@import "../../scss/typography";
```
Comment (jeherve) [inline projects/packages/search/src/dashboard/components/button/style.scss:4]: We'll probably want to move towards using the base-styles package instead.

File: projects/packages/search/src/dashboard/store/actions/jetpack-settings.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		yield updatingNotice();
+		yield setJetpackSettings( settings );
+		yield setUpdatingJetpackSettings();
+		yield updateJetpackSettingsControl( settings );
+		const updatedSettings = yield fetchJetpackSettings();
+		yield setJetpackSettings( updatedSettings );
+		return successNotice( __( 'Updated settings.' ) );
+	} catch ( e ) {
+		yield setJetpackSettings( oldSettings );
+		return errorNotice( __( 'Error Update settings…' ) );
```
Comment (jeherve) [inline projects/packages/search/src/dashboard/store/actions/jetpack-settings.js:38]:
```suggestion
		return successNotice( __( 'Updated settings.', 'jetpack' ) );
	} catch ( e ) {
		yield setJetpackSettings( oldSettings );
		return errorNotice( __( 'Error Update settings…', 'jetpack' ) );
```

File: projects/packages/search/src/dashboard/store/resolvers.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * @returns {object} - an action object.
+ */
+export function* getSearchPlanInfo() {
+	try {
+		const planInfo = yield fetchSearchPlanInfo();
+		if ( planInfo ) {
+			return setSearchPlanInfo( planInfo );
+		}
+	} catch ( e ) {
+		return errorNotice( __( 'Error fetching search plan…' ) );
```
Comment (jeherve) [inline projects/packages/search/src/dashboard/store/resolvers.js:44]:
```suggestion
		return errorNotice( __( 'Error fetching search plan…', 'jetpack' ) );
```

File: projects/packages/search/src/dashboard/store/resolvers.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * @returns {object} - an action object.
+ */
+export function* getSearchModuleStatus() {
+	try {
+		const settings = yield fetchJetpackSettings();
+		if ( settings ) {
+			return setJetpackSettings( settings );
+		}
+	} catch ( e ) {
+		return errorNotice( __( 'Error fetching settings…' ) );
```
Comment (jeherve) [inline projects/packages/search/src/dashboard/store/resolvers.js:27]:
```suggestion
		return errorNotice( __( 'Error fetching settings…', 'jetpack' ) );
```

**Review by jeherve — APPROVED** [review]
I noted things below, but nothing that cannot be fixed in follow-up PRs so I'm approving so we can move on.

File: projects/packages/search/src/dashboard/store/resolvers.js (hunk 3 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/store/resolvers.js:44]: @jeherve oops, thanks for catching this, will address in the following PR. And will see how to add the check here.

File: projects/packages/search/src/dashboard/store/resolvers.js (hunk 4 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/store/resolvers.js:27]: @jeherve oops, thanks for catching this, will address in the following PR.

File: projects/packages/search/src/dashboard/components/button/style.scss (hunk 1 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/components/button/style.scss:4]: will try to use the package in the following PR.

**Comment by kangzj** [comment]
🚢 r235709-wpcom

**Pull request merged**